source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -229,28 +259,33 @@ void AuthenticatorImpl::verifyKey() {
 
   if (!provider.forward()) {
     // Remove JWT from headers.
-    token_->removeJwt(*headers_);
+    curr_token_->removeJwt(*headers_);
   }
 
   doneWithStatus(Status::Ok);
 }
 
-bool AuthenticatorImpl::okToBypass() const {
-  // TODO(qiwzhang): use requirement field
-  return false;
-}
-
 void AuthenticatorImpl::doneWithStatus(const Status& status) {
-  ENVOY_LOG(debug, "Jwt authentication completed with: {}",
-            ::google::jwt_verify::getStatusString(status));
-  callback_->onComplete(status);
-  callback_ = nullptr;
+  // if on allow missing or failed this should verify all tokens, otherwise stop on ok.
+  if ((Status::Ok == status && !is_allow_failed_) || tokens_.empty()) {
+    tokens_.clear();
+    ENVOY_LOG(debug, "Jwt authentication completed with: {}",
+              ::google::jwt_verify::getStatusString(status));
+    callback_(status);
```

Comment:
if allow_failed,  need to convert error status to OK.

Reply:
Allow failed verifier ignores the status from authenticator and always return OK. I'll convert this status to OK and remove that behavior from the verifier.